### PR TITLE
Water temperature fix

### PR DESCRIPTION
**Pull request by SpookyTheFox**
## About The Pull Request
Stop me if you've heard this one before.
The solution freezes up into ice! x4
The solution freezes up into ice!
The solution freezes up into ice! x3
The solution freezes up into ice!

You get the idea.

Currently, if you dispense water into a container from a soda dispenser, like the one in the bar, it comes out at T0C + 0.85, which works out to 274K. If you look over in code\modules\reagents\chemistry\recipes\others.dm , you can see with /datum/chemical_reaction/ice causes water to turn to ice at this temperature. At 275K, ice turns into water. To prevent the soda dispenser from spamming the chemical reaction message and slowly freezing dispensed water into ice, I have changed it to T0C + 1.35, which works out to 274.5K, and causes neither reaction to happen. I already made a similar fix to the service borg shaker, but fixing this one was outside the scope of that PR from a while back.

Fair warning, I do not know why the old value was chosen, it may have another special purpose, but I have not yet encountered it.

## Why It's Good For The Game
This is a fix for something that spams chat and causes unintentional behavior of a chemical reaction.

## Changelog
:cl:
fix: tweaked soda dispenser reagent temperatures
/:cl:

The GBP isn't really important to me, this is a bug on a TG downstream and so I am offering my very simple fix up here for the benefit of all spacemans.

### CODE REVIEW

**Comment by maxymax13** [comment]
this is like the second time a pr like this exists. the better solution is to remove the ice reagent so i dont have to hear BLRRRBRRBRBBWWB every 5 seconds ever again

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Make a define `#define WATER_MATTERSTATE_CHANGE_TEMP 245.5` or whatever

your change here would be `dispensed_temperature = WATER_MATTERSTATE_CHANGE_TEMP`

https://github.com/tgstation/tgstation/blob/620a158e49a3aeb5be8a8141ba8039374976a2e1/code/modules/reagents/chemistry/recipes/others.dm#L788

`required_temp = WATER_MATTERSTATE_CHANGE_TEMP-0.5`

https://github.com/tgstation/tgstation/blob/620a158e49a3aeb5be8a8141ba8039374976a2e1/code/modules/reagents/chemistry/recipes/others.dm#L804

`required_temp = WATER_MATTERSTATE_CHANGE_TEMP+0.5`

never worry about this dumb bug ever again

**Comment by SpookyTheFox** [comment]
I got the bar drink dispenser too, so mixed drinks containing ice don't start the reaction either. You've all been drinking 80F/26C warm, gross drinks this entire time. Also downloaded, compiled, and manually play tested a bunch of stuff to make sure it's all working now.

**Review by SpookyTheFox — COMMENTED** [review]
Done!

File: code/__DEFINES/reagents.dm [hunk 1]
```diff
@@ -175,3 +175,5 @@
 
 #define BLASTOFF_DANCE_MOVE_CHANCE_PER_UNIT 3
 #define BLASTOFF_DANCE_MOVES_PER_SUPER_MOVE 3
+
+#define WATER_MATTERSTATE_CHANGE_TEMP 245.5 //This is the center of a 1 degree deadband in which water will neither freeze to ice nor melt to liquid
```
Comment (ExcessiveUseOfCobblestone) [inline code/__DEFINES/reagents.dm:179]:
```suggestion
///This is the center of a 1 degree deadband in which water will neither freeze to ice nor melt to liquid
#define WATER_MATTERSTATE_CHANGE_TEMP 245.5 
```

**Pull request merged**